pagefaultgames/pokerogue
# [Feature] Seasonal splash messages logic + scaffolding

## What are the changes the user will see?

Seasonal splash messages if any are active and it has messages.

## Why am I making these changes?

@damocleas wants seasonal splash messages
https://discord.com/channels/1125469663833370665/1286021055286350006

## What are the changes from a developer perspective?

_See `Files` Tab._
Made sure to add some template types to make _editors_ aware of wrong day-moth formats.

The first (example) batch of messages is:
```json
{
  "halloween": {
    "pumpkaboosAbout": "Pumpkaboos about!",
    "mayContainSpiders": "May contain spiders!",
    "spookyScaryDuskulls": "Spooky, Scary Duskulls!"
  },
  "xmas": {
    "happyHolidays": "Happy Holidays!",
    "delibirdSeason": "Delibird Season!"
  },
  "newYears": {
    "happyNewYear": "Happy New Year!"
  }
}
```

In parallel I've also optimized the way that messages are rendered to reduce unnecessary `i18next.t()` calls and also make the battles-won COUNT variable an actual i18next interpolation

### Screenshots/Videos

![image](https://github.com/user-attachments/assets/a1666b1f-0428-48ff-a634-3c9f35d6abe9)


## How to test the changes?

Make sure one of the prepared season is active.
Add some messages.
Check the console for the console.log that is being printed.
reload a few times hoping your message shows up

## Checklist
- [ ] The PR is self-contained and cannot be split into smaller PRs? => not fully. But I tried my best
